src/Symfony/Component/Form/Extension/Core/Type/ChoiceType.php

```diff
@@ -296,6 +298,16 @@ public function configureOptions(OptionsResolver $resolver)
             return $choiceTranslationDomain;
         };
 
+        $choiceFilterNormalizer = function (Options $options, $choiceFilter) {
+            if (null !== $choiceFilter && !\is_callable($choiceFilter)) {
```

Comment:
So if we pass a valid set of two values to filter out the list but the array happen to be callable this may break. Since it would add complexity to handle this edge case I suggest to either remove the support of array or document the limitation.

Reply:
not sure i understand

```php
$form = $this->createForm(ChoiceType::class, null, [
    'choices' => [1,2,3],
    'choice_filter' => [3],
]);
```

works as expected...

Reply:
@ro0NL   HashDude meant a choice_filter like this `["ClassName", "methodName"]` for this, is_callable might return true

Reply:
i see, that's edgy yes :) :+1: for removing array support from me